pyccel/parser/semantic.py

```diff
@@ -2295,6 +2296,16 @@ def _visit_FunctionDef(self, expr, **settings):
         is_private   = expr.is_private
 
         header = expr.header
+
+        keys = decorators.keys()
+        not_used = []
+        for value in keys:
+            if value not in def_decorators.__all__:
+                not_used.append(value)
```

Comment:
This can be rewritten in a more compact way (also much faster) using a list comprehension:
```python
not_used = [d for d in decorators if d not in def_decorators.__all__]
```
Please note that when you loop over a dictionary you automatically get an iterator over its keys, so there is no need to use the `keys()` method.